docs/add_data_source.md

```diff
@@ -0,0 +1,385 @@
+# How To Add Data to DLME from a New Provider
+
+When adding data to DLME from a new data provider, there are several areas of multiple codebases that must be updated. In this document, we walk through all the necessary steps and link to examples to help clarify the process.
+
+This process leverages extensions to [Traject, a configuration-based MARC to Solr indexer written in Ruby](https://github.com/traject/traject). We have extended it to provide configuration-based conversion from your data source (as XML, CSV, JSON, or binary MARC21) to the DLME MAP represented in JSON. Configuration-based conversion means each data from a new provider creates a new configuration file to indicate their mappings and any fixed fields for the dataset. The mappings in the configurations are meant to be as simple as possible so that providers don't need to be Ruby programmers to write these. To read more about our extensions to Traject for this data pipeline, you can read our [Traject Extensions documentation](extensions_to_traject.md).
+
+## Steps
+
+1. [Load Your Metadata into Our DLME Metadata GitHub Repository](#1-stage-your-metadata)
+2. [Create a Branch on This GitHub Repository](#2-create-a-dlme-github-repository-branch)
+3. [Add Your Data Source's Settings](#3-add-your-data-source-settings)
+4. [Establish data mapping](#4-create-your-data-mapping)
+5. [Write mapping configuration](#5-create-your-mapping-configuration-file)
+6. [Optional: Write mapping macro for data transformations](#6-mapping-macros)
+7. [Write spec to test data mapping and provide fixture data for testing](#7-test-your-data-mapping)
+8. [Create Your Pull Request](#8-create-a-pull-request)
+
+## 1. Stage Your Metadata
+
+At present, this DLME prototype loads metadata from this DLME Metadata GitHub repository: https://github.com/waynegraham/dlme-metadata. You will need to add your metadata - in the appropriate directory - via a pull request to that DLME Metadata GitHub repository for it to be loaded by our pipeline.
```

Comment:
are we really using that repo, or the fork here?  https://github.com/sul-dlss/dlme-metadata/